packages/chain-controller/src/SnapHandlerClient.ts

```diff
@@ -0,0 +1,130 @@
+import type { JsonRpcRequest } from '@metamask/keyring-api/dist/JsonRpcRequest';
+import type { SnapController } from '@metamask/snaps-controllers';
+import type { SnapId } from '@metamask/snaps-sdk';
+import { HandlerType } from '@metamask/snaps-utils';
+import type { Json } from '@metamask/utils';
+import { v4 as uuid } from 'uuid';
+
+/**
+ * Handler for Snap requests.
+ */
+export type Handler = SnapController['handleRequest'];
+
+/**
+ * Send requests to a Snap through a Snap request handler.
+ */
+class SnapHandlerSender {
+  #snapId: SnapId;
+
+  #origin: string;
+
+  #handler: Handler;
+
+  #handlerType: HandlerType;
+
+  /**
+   * Constructor for `SnapHandlerSender`.
+   *
+   * @param handler - The Snap request handler to send requests to.
+   * @param handlerType - The handler type.
+   * @param snapId - The ID of the snap to use.
+   * @param origin - The sender's origin.
+   */
+  constructor(
+    handler: Handler,
+    handlerType: HandlerType,
```

Comment:
I guess so yes. Right now, it's not yet available on Snaps side, and since this class is mostly generic I just wanted to use "any handler".

This is highly inspired from the `keyring-api` clients like this one:
- https://github.com/MetaMask/keyring-api/blob/main/src/KeyringSnapRpcClient.ts